Source/AwsCommonRuntimeKit/auth/credentials/CRTCredentialsProvider.swift

```diff
@@ -68,24 +65,19 @@ class WrappedCRTCredentialsProvider {
 
     static func setUpShutDownOptions(shutDownOptions: CRTCredentialsProviderShutdownOptions?)
     -> aws_credentials_provider_shutdown_options {
-        let shutDownOptionsC: aws_credentials_provider_shutdown_options?
-        if let shutDownOptions = shutDownOptions {
-
-            let pointer = UnsafeMutablePointer<CRTCredentialsProviderShutdownOptions>.allocate(capacity: 1)
-            pointer.initialize(to: shutDownOptions)
-            shutDownOptionsC = aws_credentials_provider_shutdown_options(shutdown_callback: { userData in
-                guard let userData = userData else {
-                    return
-                }
-                let pointer = userData.assumingMemoryBound(to: CRTCredentialsProviderShutdownOptions.self)
-                defer {pointer.deinitializeAndDeallocate()}
-                pointer.pointee.shutDownCallback()
-
-            }, shutdown_user_data: pointer)
-        } else {
-            shutDownOptionsC = aws_credentials_provider_shutdown_options()
-        }
-        return shutDownOptionsC!
+
+        let pointer: UnsafeMutablePointer<CRTCredentialsProviderShutdownOptions>? = fromOptionalPointer(ptr: shutDownOptions)
+        let shutDownOptionsC = aws_credentials_provider_shutdown_options(shutdown_callback: { userData in
+            guard let userData = userData else {
+                return
+            }
+            let pointer = userData.assumingMemoryBound(to: CRTCredentialsProviderShutdownOptions.self)
+            defer {pointer.deinitializeAndDeallocate()}
+            pointer.pointee.shutDownCallback()
+
+        }, shutdown_user_data: pointer)
```

Comment:
Feels like this section needs a format/lint

Reply:
formatted and linted